web/src/views/FileBrowserView/FileBrowser.vue

```diff
@@ -377,18 +377,40 @@ const isOwner = computed(() => !!(
 ));
 const itemsNotFound = computed(() => items.value && !items.value.length);
 
+function replaceStringsInEndpoint(
+  endpoint: string,
+  substitutions: {[key: string]: string} | undefined,
+) {
+  if (!substitutions) return endpoint;
+  let result = endpoint;
+  Object.entries(substitutions).forEach(([key, value]) => {
+    result = result.replace(key, value);
+  });
+  return result;
+}
+
+const isStaging = process.env.VUE_APP_SENTRY_ENVIRONMENT === 'staging';
+
 function getExternalServices(path: AssetPath) {
   const servicePredicate = (service: Service, _path: AssetPath) => (
     new RegExp(service.regex).test(path.path)
           && _path.asset !== null
           && _path.aggregate_size <= service.maxsize
   );
 
+  const baseApiUrl = isStaging
+    ? 'https://api-staging.dandiarchive.org'
+    : 'https://api.dandiarchive.org';
+  const substitutions = path.asset ? {
+    $s3_url$: trimEnd(path.asset.url, '/'),
+    $api_download_url$: `${baseApiUrl}/api/assets/${path.asset.asset_id}/download/`,
```

Comment:
Those are both URLs to the asset download. So I think we could/should make them a little more uniform, what about

```suggestion
    $asset_s3_download_url$: trimEnd(path.asset.url, '/'),
    $asset_api_download_url$: `${baseApiUrl}/api/assets/${path.asset.asset_id}/download/`,
```
so in the future we might expand the listing here without fear of collision?

Reply:
BTW , in https://github.com/dandi/dandi-archive/pull/1655 (does this PR supersedes it?) we seems also wanted to expose `asset_id`.  I think it would be safe to include it here as well but I guess we could add more as demand comes.

Reply:
Thanks @yarikoptic !

I have adjusted the strings as you suggested and added a $asset_id$ (which of course is unused at this point)

Yes, I think this is accomplishing the idea laid out in #1655